threedi_schema_edits/breach_location.py

```diff
@@ -0,0 +1,484 @@
+# -*- coding: utf-8 -*-
+# (c) Nelen & Schuurmans, see LICENSE.rst.
+
+import logging
+import itertools
+import collections
+
+from qgis.core import QgsFeatureRequest
+from qgis.core import QgsFeature
+from qgis.core import QgsGeometry
+from qgis.core import QgsVectorLayer
+from qgis.core import QgsPoint
+from qgis.core import QgsField
+
+from PyQt4.QtCore import QVariant
+
+from ThreeDiToolbox.utils.geo_utils import calculate_perpendicular_line
+from ThreeDiToolbox.utils.geo_utils import get_distance
+from ThreeDiToolbox.utils.geo_utils import get_epsg_code_from_layer
+from ThreeDiToolbox.utils.geo_utils import get_extrapolated_point
+
+from ThreeDiToolbox.utils.utils import parse_db_source_info
+
+from ThreeDiToolbox.threedi_schema_edits.predictions import Predictor
+from ThreeDiToolbox.utils import constants
+from ThreeDiToolbox.utils import utils
+
+log = logging.getLogger(__name__)
+
+EXTRAPLORATION_RATIO = 20
+
+
+class BreachLocation(object):
```

Comment:
A description on what this class does would be helpful, since this class is pretty big.